wadsrc/static/zscript/doom/bossbrain.txt

```diff
@@ -165,23 +165,24 @@ extend class Actor
 			boom.tics -= random[BrainScream](0, 7);
 			if (boom.tics < 1) boom.tics = 1;
 		}
+		else boom.Destroy();
```

Comment:
This would trigger a VM abort if ```boom``` was null.

Reply:
I don't think it will actually trigger a VM abort in-game, as far as I know.

Reply:
Try this simple example:

```
class BossBrain2 : BossBrain replaces BossBrain
{
	States
	{
	Death:
		BBRN A 100
		{
			string type = "BlahBoom";
			A_BrainScream(type);
		}
		BBRN AA 10;
		BBRN A -1 A_BrainDie;
		Stop;
	}
}
```
Since `BlahBoom` doesn't exist as a class, the spawning fails. And yes, strings can be used here without problems.

Reply:
Wrong thing to argue about.

Reply:
What?